src/main/kotlin/no/nav/amt/person/service/nav_bruker/NavBrukerService.kt

```diff
@@ -90,18 +90,23 @@ class NavBrukerService(
 
 	fun oppdaterKontaktinformasjon(personer: List<Person>) {
 		personer.forEach {person ->
-			repository.finnBrukerId(person.personIdent)?.let { brukerId ->
-				val kontaktinformasjon = krrProxyClient.hentKontaktinformasjon(person.personIdent)
-				val pdlTelefon = pdlClient.hentTelefon(person.personIdent)
-
-				repository.oppdaterKontaktinformasjon(
-					navBrukerId = brukerId,
-					telefon = kontaktinformasjon.telefonnummer ?: pdlTelefon,
-					epost = kontaktinformasjon.epost,
-				)
-			}
+			oppdaterKontaktinformasjon(person.personIdent)
 		}
+	}
+
+	private fun oppdaterKontaktinformasjon(personIdent: String) {
+		val eksisterendeKontaktinfo = repository.finnKontaktinformasjon(personIdent) ?: return
```

Comment:
Bør vi ikke prøve å oppdatere hvis vi mangler kontaktinformasjon?

Reply:
Jo, dette er nok ikke tydelig nok skrevet. `finnKontaktinformajson` gir oss null hvis det ikke finnes noen bruker med denne `personIdenten`. Jeg skal fundere litt på om det er en mer forklarende måte å skrive det på.

Jeg er mer usikker på hvordan jeg ønsker å håndtere linjen nedenfor (100-103). Jeg tror det er greit å ikke oppdatere hvis KRR feiler med alt annet enn 404, men jeg klarte ikke å se en god måte å løse det på, kanskje klienten kan returnere et `NoSuchElementException` eller et tomt kontaktinfo objekt...

Reply:
Aha, skjønner. `finnKontaktinformasjonHvisPersonFinnes` blir kanskje i lengste laget? 😅 

Hmm. Vil vi forsøke å oppdatere senere hvis det feiler? Hvis vi bare prøver igjen senere så tenker jeg det er helt greit sånn det er nå, altså at vi prøver, og kommer det en eller annen feil ved oppslag mot KRR, så prøver vi bare igjen. Når vi logger error så vil vi vel få alarmer på det, så da vil vi vel oppdage det hvis det er noen brukere vi aldri vil få oppdatert kontaktinfo for? Eller overser jeg noe her nå..? :)

Reply:
Hehe, det er i det minste mer tydelig enn hva jeg har kommet på så langt 👍 

Det kan hende at det er greit som det er for nå, som du sier så er det kankjse bedre å vente å se om det feiler først. Jeg vet ikke helt hvor ofte eller i hvilke tilfeller man vil få 404 fra KRR i prod, men i dev så får vi det på noen test-personer bestandig, og da er det kanskje heller ønskelig at vi sjekker mot pdl, ellers så vil de aldri bli oppdatert slik det er nå.

Reply:
Det ser ut som at KRR alltid skal gi svar, bortsett fra hvis personen ikke finnes i PDL (da får man 404): https://github.com/navikt/digdir-krr, https://digdir-krr-proxy.intern.dev.nav.no/swagger-ui/index.html?configUrl=/v3/api-docs/swagger-config#/person-controller/getPerson

At vi har personer i løsningen vår som ikke finnes i PDL skal aldri kunne skje i prod, men i dev kan det jo skje. Kanskje vi heller bør ha et unntak hvis cluster er dev-gcp at hvis det feiler så går vi bare videre uten å prøve på nytt?

Reply:
Aha! Bra spotta 🏆 

Kanskje man kan la være å logge errors i dev, når det skjer ja. Hvis vi ønsker å følge med på alarmer i dev.